class Solution {
    public boolean isHappy(int n) {
        Set<Integer> set = new HashSet<>();
        int digit;
        while(n!=1){
            if(set.contains(n)){
                return false;
            }
            set.add(n);
            int sum = 0;
            while(n>0){
            digit = n%10;
            sum += digit*digit;
            n = n/10;
    }
    n = sum;
        }
        return true;
    }
}